python/lsst/ts/mtaos/mtaos.py

```
@@ -1273,11 +1275,37 @@ async def run_closed_loop(self) -> None:
 
                         config_yaml = yaml.safe_dump(config)
 
-                        await self._execute_ofc(
-                            userGain=gain,
-                            config=config_yaml,
-                            timeout=self.CMD_TIMEOUT,
-                        )
+                        try:
+                            await self._execute_ofc(
+                                userGain=gain,
+                                config=config_yaml,
+                                timeout=self.CMD_TIMEOUT,
+                            )
+                            ofc_failure_count = 0
+                        except Exception:
+                            ofc_failure_count += 1
+                            self.log.warning(
+                                "OFC execution failed. Failures: "
+                                f"{ofc_failure_count}/{self.max_ofc_consecutive_failures}."
+                            )
+                            if ofc_failure_count >= self.max_ofc_consecutive_failures:
+                                await self.fault(
```

you should just raise the exception here.